File: goal-keeper/src/data/dto/SessionTeamResultDto.ts
import {JsonObject, JsonProperty} from 'json2typescript';
import PlayerDto from './PlayerDto';
import {SessionResult} from "@/data/enum/SessionResult";

@JsonObject('SessionTeamResultDto')
export default class SessionTeamResultDto {

    @JsonProperty('id', String)
    public id: string | undefined = undefined;

    @JsonProperty('playersScored', [PlayerDto])
    public playersScored: PlayerDto[] | undefined = undefined;

    @JsonProperty('result', String)
    public result: SessionResult | undefined = undefined;

    public static create(object: SessionTeamResultDto) {
        const instance = new SessionTeamResultDto();
        instance.id = object.id;
        instance.playersScored = object.playersScored;
        instance.result = object.result;
        return instance;
    }
}

File: goal-keeper-backend/src/main/java/com/goalkeeper/api/dto/SessionTeamResultDto.java
package com.goalkeeper.api.dto;

import com.fasterxml.jackson.annotation.JsonRootName;
import com.goalkeeper.api.domain.Session;
import com.goalkeeper.api.domain.SessionTeam;
import com.goalkeeper.api.domain.SessionTeamResult;
import com.goalkeeper.api.domain.SessionTeamResult.SessionResult;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;
import java.util.UUID;

@Builder
@Data
@JsonRootName("sessionTeamResult")
@NoArgsConstructor
@AllArgsConstructor
public class SessionTeamResultDto {

    private UUID publicId;
    private List<PlayerDto> playersScored;
    private SessionResult result;
    private SessionTeamDto sessionTeam;
}
